crates/cairo-types/src/types/data_type.rs

```diff
@@ -3,106 +3,78 @@ use crate::Result;
 use alloc::{format, vec, vec::Vec};
 use starknet::core::types::FieldElement;
 
-impl CairoType for FieldElement {
-    type RustType = Self;
-
-    fn serialize(rust: &Self::RustType) -> Vec<FieldElement> {
-        vec![*rust]
-    }
-
-    fn deserialize(felts: &[FieldElement], offset: usize) -> Result<Self::RustType> {
-        Ok(felts[offset])
-    }
-}
-
-impl CairoType for bool {
-    type RustType = Self;
+macro_rules! implement_trait_for_unsigned {
+    ($type:ty) => {
+        impl CairoType for $type {
+            type RustType = Self;
 
-    fn serialize(rust: &Self::RustType) -> Vec<FieldElement> {
-        vec![FieldElement::from(*rust as u32)]
-    }
+            fn serialize(rust: &Self::RustType) -> Vec<FieldElement> {
+                vec![FieldElement::from(*rust)]
+            }
 
-    fn deserialize(felts: &[FieldElement], offset: usize) -> Result<Self::RustType> {
-        if felts[offset] == FieldElement::ONE {
-            Ok(true)
-        } else {
-            Ok(false)
+            fn deserialize(felts: &[FieldElement], offset: usize) -> Result<Self::RustType> {
+                let temp: u128 = felts[offset].try_into().unwrap();
+                Ok(temp as $type)
+            }
         }
-    }
-}
-
-// TODO: write a macro to impl the trait for uints and ints.
-impl CairoType for u8 {
-    type RustType = Self;
-
-    fn serialize(rust: &Self::RustType) -> Vec<FieldElement> {
-        vec![FieldElement::from(*rust)]
-    }
-
-    fn deserialize(felts: &[FieldElement], offset: usize) -> Result<Self::RustType> {
-        Ok(felts[offset].try_into().unwrap())
-    }
+    };
 }
 
-impl CairoType for u16 {
-    type RustType = Self;
+macro_rules! implement_trait_for_signed {
+    ($type:ty) => {
+        impl CairoType for $type {
+            type RustType = Self;
 
-    fn serialize(rust: &Self::RustType) -> Vec<FieldElement> {
-        vec![FieldElement::from(*rust)]
-    }
+            fn serialize(rust: &Self::RustType) -> Vec<FieldElement> {
+                vec![FieldElement::from(*rust as usize)]
+            }
 
-    fn deserialize(felts: &[FieldElement], offset: usize) -> Result<Self::RustType> {
-        Ok(felts[offset].try_into().unwrap())
-    }
+            fn deserialize(felts: &[FieldElement], offset: usize) -> Result<Self::RustType> {
+                let temp: u16 = felts[offset].try_into().unwrap();
```

Comment:
```suggestion
                let temp: u128 = felts[offset].try_into().unwrap();
```
Tests pass because you used 123, but actually if you change for higher values for signed ints, tests fail.